src/Uno.UI/Microsoft/UI/Xaml/Controls/Repeater/ItemsSourceView.cs

```diff
@@ -0,0 +1,312 @@
+		//					{
+		//		observableCollection.VectorChanged(m_eventToken);
+		//	}
+		//}
+
+		void ListenToCollectionChanges()
+		{
+			if (m_vector == null)
+			{
+				throw new InvalidOperationException("Backing vector not set");
+			}
+			var incc = m_vector as INotifyCollectionChanged;
+			if (incc != null)
+			{
+				m_eventToken = incc.CollectionChanged({ this, &OnCollectionChanged });
+				m_notifyCollectionChanged = incc;
+			}
+			else
+			{
+				var bindableObservableVector = m_vector.try_as<IBindableObservableVector>();
+				if (bindableObservableVector)
+				{
+					m_eventToken = bindableObservableVector.VectorChanged({ this, &OnBindableVectorChanged });
+					m_bindableObservableVector.set(bindableObservableVector);
+				}
+				else
+				{
+					var observableVector = m_vector.try_as<IObservableVector<object>>();
+					if (observableVector)
+					{
+						m_eventToken = observableVector.VectorChanged({ this, &OnVectorChanged });
+						m_observableVector.set(observableVector);
+					}
+				}
+			}
+		}
+
+		private void OnCollectionChanged(
+					object sender,
+					NotifyCollectionChangedEventArgs e)
+		{
+			OnItemsSourceChanged(e);
+		}
+
+		void OnBindableVectorChanged(IBindableObservableVector sender, object e)
+		{
+			OnVectorChanged(
+				sender as IObservableVector<object>,
+				e as IVectorChangedEventArgs);
+		}
+
+		void OnVectorChanged(
+			IObservableVector<object> sender,
+			IVectorChangedEventArgs e)
+		{
+			// We need to build up NotifyCollectionChangedEventArgs here to raise the event.
+			// There is opportunity to make this faster by caching the args if it does 
+			// show up as a perf issue.
+			// Also note that we do not access the data - we just add nullptr. We just 
+			// need the count.
+
+			NotifyCollectionChangedAction action;
+			int oldStartingIndex = -1;
+			int newStartingIndex = -1;
+
+			var oldItems = new List<object>();
+			var newItems = new List<object>();
+
+			switch (e.CollectionChange)
+			{
+				case CollectionChange.ItemInserted:
+					action = NotifyCollectionChangedAction.Add;
+					newStartingIndex = (int)e.Index;
+					newItems.Append(null);
+					break;
+				case CollectionChange.ItemRemoved:
+					action = NotifyCollectionChangedAction.Remove;
+					oldStartingIndex = (int)e.Index;
+					oldItems.Append(null);
+					break;
+				case CollectionChange.ItemChanged:
+					action = NotifyCollectionChangedAction.Replace;
+					oldStartingIndex = (int)e.Index;
+					newStartingIndex = oldStartingIndex;
+					newItems.Append(null);
+					oldItems.Append(null);
+					break;
+				case CollectionChange.Reset:
+					action = NotifyCollectionChangedAction.Reset;
+					break;
+				default:
+					throw new InvalidOperationException("Unsupported collection change");
+			}
+
+			//OnItemsSourceChanged(
+			//	new NotifyCollectionChangedEventArgs(
+			//		action,
+			//		newItems,
+			//		oldItems,
+			//		newStartingIndex,
+			//		oldStartingIndex);
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Remove this commented out code.](https://app.codacy.com/gh/unoplatform/uno/pullRequest?prid=5156819)